lib/Doctrine/ORM/Cache/QueryCacheKey.php

```diff
@@ -45,14 +45,27 @@ class QueryCacheKey extends CacheKey
     public $cacheMode;
 
     /**
-     * @param string  $hash      Result cache id
-     * @param integer $lifetime  Query lifetime
-     * @param integer $cacheMode Query cache mode
+     * READ-ONLY: Public only for performance reasons, it should be considered immutable.
+     *
+     * @var TimestampCacheKey|null
+     */
+    public $timestampKey;
+
+    /**
+     * @param string $hash Result cache id
+     * @param integer $lifetime Query lifetime
+     * @param int $cacheMode Query cache mode
+     * @param TimestampCacheKey|null $timestampKey
```

Comment:
Basically because of https://github.com/doctrine/doctrine2/blob/8a87fa2d01a01f4ae939559b10cb5733132930cf/tests/Doctrine/Tests/ORM/Functional/SecondLevelCacheQueryCacheTest.php#L1106 and https://github.com/doctrine/doctrine2/blob/8a87fa2d01a01f4ae939559b10cb5733132930cf/tests/Doctrine/Tests/ORM/Functional/SecondLevelCacheQueryCacheTest.php#L1117

Since the ResultSetMapping don't give the entity name.